NSS-Day-Cohort-42/rare-API-media-moguls
# Sm updateposts

#### Changes Made
1. Added `update` method to `category.py` in views. 

#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout sm-updateposts
```

2. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py runserver
```
3.  Test app functionality.
> Go to postman and run a GET with `http://localhost:8000/posts` and ensure that you can get all posts
>Change the method to PUT and change the URL to `http://localhost:8000/posts/1` to update the first post. 
>In the BODY tab, input the following object:
`    {
        "id": 1,
        "title": "BARKSGIVING APPROACHES",
        "publication_date": "2020-08-18",
        "content": "PROOF that your heckin pupper is the only reason your family invites you to Thanksgiving. (so you better bring him)",
        "rareuser": 1,
        "category": 2
    }`
>Check that you receive a 204 status. 
>GET all posts again and check that the first post has been updated!


4. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
